extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -614,14 +626,20 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		}
 	}
 
-	getTreeItem(element: BookTreeItem): vscode.TreeItem {
-		return element;
+	async getTreeItem(element: BookTreeItem): Promise<vscode.TreeItem> {
+		let bookItem: BookTreeItem = undefined;
+		let notebookPath = element.tooltip;
+		let bookOfElement = this.books.length > 1 ? this.books.find(b => notebookPath.indexOf(b.bookPath) > -1) : this.currentBook;
+		if (bookOfElement?.bookItems.indexOf(element) > -1) {
+			bookItem = bookOfElement?.bookItems.find(b => b === element);
+		}
+		return bookItem;
 	}
 
 	getChildren(element?: BookTreeItem): Thenable<BookTreeItem[]> {
 		if (element) {
 			if (element.sections) {
-				return Promise.resolve(this.currentBook.getSections(element.tableOfContents, element.sections, element.root, element.book).then(sections => { return sections; }));
+				return Promise.resolve(this.currentBook.getSections(element).then(sections => { return sections; }));
```

Comment:
The .then doesn't appear to be doing anything, you should be able to just return the Promise.resolve